wandb/data_types.py

```diff
@@ -663,6 +667,52 @@ def transform(table):
         return {"_type": "table", "columns": table.columns, "data": table.rows[:Table.MAX_ROWS]}
 
 
+class Audio(object):
+    MAX_AUDIO_COUNT = 100
+
+    def __init__(self, data, sample_rate=None, caption=None):
+        """
+        Accepts numpy array of audio data. 
+        """
+        self.audio_data = data
+        self.sample_rate = sample_rate
+        self.caption = caption
+
+    @staticmethod
+    def transform(audio_list, out_dir, key, step):
+        if len(audio_list) > Audio.MAX_AUDIO_COUNT:
+            logging.warn(
+                "The maximum number of audio files to store per step is %i." % Audio.MAX_AUDIO_COUNT)
+        sf = util.get_module(
+            "soundfile", required="wandb.Audio requires the soundfile package. To get it, run: pip install soundfile")
+        base_path = os.path.join(out_dir, "media", "audio")
+        util.mkdir_exists_ok(base_path)
+        for i, audio in enumerate(audio_list[:Audio.MAX_AUDIO_COUNT]):
+            sf.write(os.path.join(base_path, "{}_{}_{}.wav".format(key, step, i)), audio.audio_data, audio.sample_rate)
+        meta = {"_type": "audio", "count": min(len(audio_list), Audio.MAX_AUDIO_COUNT)}
+        sample_rates = Audio.sample_rates(audio_list[:Audio.MAX_AUDIO_COUNT])
+        if sample_rates:
+            meta["sampleRates"] = sample_rates
+        captions = Audio.captions(audio_list[:Audio.MAX_AUDIO_COUNT])
+        if captions:
+            meta["captions"] = captions
+        return meta
+
+    @staticmethod
+    def sample_rates(audio_list):
+        if audio_list[0].sample_rate != None:
```

Comment:
This seems a little weird. Why do you only care about the first Audio in the list? Why not just always return the list of all sample rates?

Reply:
@adrianbg Is this better?  https://github.com/wandb/client/pull/248/commits/42e2cc81fec02dcf49fdc2c96353d0f7515cd2e1